### PR TITLE
Jg us17 items index

### PR DESCRIPTION
**Pull request by elguapogordo**
test and function to add features:
  * all users see only active items on items index
  * item image is now also a link to item show
test and function for item model:
  * active items logic is now in the model
  * added test for active_item method to model
minor refactoring

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def index
       @merchant = Merchant.find(params[:merchant_id])
       @items = @merchant.items
     else
-      @items = Item.all
+      @items = Item.where(active?: true)
```
Comment (elguapogordo) [inline app/controllers/items_controller.rb]: activerecord filter to ignore inactive items

File: app/views/items/index.html.erb [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
+      <h2><%=link_to item.name, "/items/#{item.id}" %></h2>
       <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
-      <img src= <%= item.image %>>
-      <p> <%= item.description unless @merchant%> </p>
-      <p>Price: <%=number_to_currency(item.price) %> </p>
-      <p>Inventory: <%= item.inventory %> </p>
+      <%=link_to image_tag("#{item.image}"), "/items/#{item.id}" %>
+      <p> <%= item.description unless @merchant%></p>
+      <p>Price: <%=number_to_currency(item.price) %></p>
+      <p>Inventory: <%= item.inventory %></p>
```
Comment (elguapogordo) [inline app/views/items/index.html.erb]: there were some weird spaces between the formatting. those were removed.

File: app/views/items/index.html.erb [hunk 3]
```diff
@@ -7,12 +7,12 @@
...
   <% @items.each do |item| %>
     <section class = "grid-item" id= 'item-<%=item.id%>'>
-      <h2> <%=link_to item.name, "/items/#{item.id}" %> </h2>
+      <h2><%=link_to item.name, "/items/#{item.id}" %></h2>
       <p>Sold by: <%=link_to item.merchant.name, "/merchants/#{item.merchant.id}" %></p>
-      <img src= <%= item.image %>>
-      <p> <%= item.description unless @merchant%> </p>
-      <p>Price: <%=number_to_currency(item.price) %> </p>
-      <p>Inventory: <%= item.inventory %> </p>
+      <%=link_to image_tag("#{item.image}"), "/items/#{item.id}" %>
```
Comment (elguapogordo) [inline app/views/items/index.html.erb]: image is now a link to item show

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (zachholcomb) [reply app/controllers/items_controller.rb]: like the use of active record here, we should move this to the Item model

Comment (steve-meyers) [reply app/controllers/items_controller.rb]: Should we move this ActiveRecord login into the model to follow proper MVC?

**Review by jrsewell400 — CHANGES_REQUESTED** [review]
Move ".where(active? = true) into model method

File: spec/features/items/index_spec.rb [hunk 4]
```diff
@@ -47,15 +47,13 @@
...
-        expect(page).to have_css("img[src*='#{@dog_bone.image}']")
-      end
+        expect(page).to have_no_link(@dog_bone.name)
+        expect(page).to have_no_content(@dog_bone.description)
+        expect(page).to have_no_content("Price: $#{@dog_bone.price}")
+        expect(page).to have_no_content("Inactive")
+        expect(page).to have_no_content("Inventory: #{@dog_bone.inventory}")
+        expect(page).to have_no_css("img[src*='#{@dog_bone.image}']")
+
     end
```
Comment (steve-meyers) [inline spec/features/items/index_spec.rb:59]: Should we check that the links of this page are going to the right places?

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (elguapogordo) [reply app/controllers/items_controller.rb]: moved to model

File: app/models/item.rb [hunk 5]
```diff
@@ -12,6 +12,9 @@ class Item <ApplicationRecord
   validates_inclusion_of :active?, :in => [true, false]
   validates_numericality_of :price, greater_than: 0
 
+  def self.active_items
+    where(active?: true)
```
Comment (zachholcomb) [inline app/models/item.rb:16]: this is nice! follows mvc well

**Review by jrsewell400 — APPROVED** [review]

File: spec/features/items/index_spec.rb [hunk 6]
```diff
@@ -12,18 +12,20 @@
...
 
       expect(page).to have_link(@tire.name)
       expect(page).to have_link(@tire.merchant.name)
+      expect(page).to have_css("img[src*='#{@tire.image}']")
       expect(page).to have_link(@pull_toy.name)
       expect(page).to have_link(@pull_toy.merchant.name)
-      expect(page).to have_link(@dog_bone.name)
-      expect(page).to have_link(@dog_bone.merchant.name)
+      expect(page).to have_css("img[src*='#{@pull_toy.image}']")
+      find(:xpath, "//a/img[@alt='Tug toy dog pull 9010 2 800x800']/..").click
```
Comment (zachholcomb) [inline spec/features/items/index_spec.rb:24]: good job figuring out how to click on image link

**Comment by jrsewell400** [comment]
Great job refactoring and updating active items to model method Jesse!  Merging.

File: spec/features/items/index_spec.rb (hunk 6 above)
Comment (steve-meyers) [reply spec/features/items/index_spec.rb:24]: ❤️

File: app/controllers/items_controller.rb [hunk 7]
```diff
@@ -5,7 +5,7 @@ def index
       @merchant = Merchant.find(params[:merchant_id])
       @items = @merchant.items
     else
-      @items = Item.all
+      @items = Item.active_items
```
Comment (elguapogordo) [inline app/controllers/items_controller.rb:8]: logic removed from controller, added to model

File: spec/features/items/index_spec.rb (hunk 6 above)
Comment (elguapogordo) [inline spec/features/items/index_spec.rb:24]: this is the test i added to verify image link

File: spec/features/items/index_spec.rb (hunk 4 above)
Comment (elguapogordo) [reply spec/features/items/index_spec.rb:59]: yes, good call and that has been added now, as well

**Pull request merged**